### PR TITLE
Fix get inventory position not correctly at Pipeline

### PR DESCRIPTION
**Pull request by azexcy**
Changes proposed in this pull request:
  - Fix get inventory position not correctly at Pipeline

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: kernel/data-pipeline/core/src/test/java/org/apache/shardingsphere/data/pipeline/common/job/progress/InventoryIncrementalJobItemProgressTest.java [hunk 1]
```diff
@@ -76,6 +79,24 @@ void assertGetIncrementalDataLatestActiveTimeMillis() {
...
+        progresses.put("ds.test_order#0", new InventoryTaskProgress(new FinishedPosition()));
+        JobItemInventoryTasksProgress progress = new JobItemInventoryTasksProgress(progresses);
+        Map<String, IngestPosition> orderPosition = progress.getInventoryPosition("order");
+        assertThat(orderPosition.size(), is(1));
+        assertTrue(orderPosition.containsKey("ds.order#0"));
+        Map<String, IngestPosition> testOrderPosition = progress.getInventoryPosition("test_order");
+        assertThat(testOrderPosition.size(), is(1));
+        assertTrue(testOrderPosition.containsKey("ds.test_order#0"));
+        Map<String, IngestPosition> orderItemPosition = progress.getInventoryPosition("order_item");
+        assertThat(orderItemPosition.size(), is(2));
```
Comment (sandynz) [inline kernel/data-pipeline/core/src/test/java/org/apache/shardingsphere/data/pipeline/common/job/progress/InventoryIncrementalJobItemProgressTest.java:97]: Could we assert position but not only size?

File: kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/common/job/progress/JobItemInventoryTasksProgress.java [hunk 2]
```diff
@@ -43,7 +43,7 @@ public final class JobItemInventoryTasksProgress {
      * @return inventory position
      */
     public Map<String, IngestPosition> getInventoryPosition(final String tableName) {
-        Pattern pattern = Pattern.compile(String.format("%s(#\\d+)?", tableName));
+        Pattern pattern = Pattern.compile(String.format("\\.%s#(\\d+)?", tableName));
```
Comment (sandynz) [inline kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/common/job/progress/JobItemInventoryTasksProgress.java:46]: Is `#` always existing?

From old code:
```
    private String generateTaskId(final InventoryDumperConfiguration inventoryDumperConfig) {
        String result = String.format("%s.%s", inventoryDumperConfig.getDataSourceName(), inventoryDumperConfig.getActualTableName());
        return null == inventoryDumperConfig.getShardingItem() ? result : result + "#" + inventoryDumperConfig.getShardingItem();
    }
```

**Review by sandynz — CHANGES_REQUESTED** [review]

File: kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/common/job/progress/JobItemInventoryTasksProgress.java (hunk 2 above)
Comment (azexcy) [reply kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/common/job/progress/JobItemInventoryTasksProgress.java:46]: If using `InventoryTaskSplitter`, the shardingItem wouldn't be null, scenario not found where shardingItem is null

**Review by sandynz — APPROVED** [review]

File: kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/common/job/progress/JobItemInventoryTasksProgress.java (hunk 2 above)
Comment (sandynz) [reply kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/common/job/progress/JobItemInventoryTasksProgress.java:46]: OK

**Pull request merged**